src/containers/Registration.tsx

```diff
@@ -96,110 +96,126 @@ const Registration = () => {
+          <Form.Control
+            {...register("username")}
+            className="form-control"
+            type="text"
+            name="username"
+            placeholder={t("validation:fields.fullname")}
+            maxLength={50}
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.fullname")}
+          </StyledLabel>
+          <StyledLabel
+            counter
+            className="fs-6 mt-2 text-end"
+            htmlFor="floatingInputCustom"
+          >
+            {watchusername ? watchusername.length : 0} / 50
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic fs-6 mt-1">
+          {errors.username &&
+            errors.username.message &&
+            t(`${errors.username.message}`)}
+        </p>
+        <StyledFormFloating className="form-floating mt-3">
+          <Form.Control
+            className="form-control"
+            {...register("email")}
+            type="text"
+            name="email"
+            placeholder={t("validation:fields.email")}
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.email")}
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic mt-1 fs-6">
+          {errors.email && errors.email.message && t(`${errors.email.message}`)}
+        </p>
+        <StyledFormFloating className="form-floating mt-3">
+          <Form.Control
+            className="fs-6"
+            {...register("dateOfBirth")}
+            type="date"
+            name="dateOfBirth"
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.dateOfBirth")}
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic mt-1 fs-6">
+          {errors.dateOfBirth &&
+            errors.dateOfBirth.message &&
+            t(`${errors.dateOfBirth.message}`)}
+        </p>
+        <StyledFormFloating className="form-floating mt-3">
+          <Form.Control
+            className="form-control"
+            {...register("password")}
+            type="password"
+            name="password"
+            placeholder={t("validation:fields.password")}
+            autoComplete="on"
+          />
+          <StyledLabel
+            className="pt-1 text-muted"
+            htmlFor="floatingInputCustom"
+          >
+            {t("validation:fields.password")}
+          </StyledLabel>
+        </StyledFormFloating>
+        <p className="text-danger fst-italic mt-1 fs-6">
+          {errors.password &&
+            errors.password.message &&
+            t(`${errors.password.message}`)}
+        </p>
```

Comment:
Use ternary operators in the error display.
Add i18 to the registration shema, like the login shema.